src/metabase/models/interface.clj

```diff
@@ -514,12 +529,14 @@
     (not (:created_at obj)) (assoc :created_at (now))))
 
 (defn- add-updated-at-timestamp [obj]
-  ;; don't stomp on `:updated_at` if it's already explicitly specified.
-  (let [changes-already-include-updated-at? (if (t2/instance? obj)
-                                              (:updated_at (t2/changes obj))
-                                              (:updated_at obj))]
+  (let [changed-fields (set (keys (if (t2/instance obj)
+                                    (t2/changes obj)
+                                    (:updated_at obj))))
```

Comment:
I'm not sure what happens if this branch is hit? Wouldn't we call set and keys on the value at :updated_at?

Reply:
Maybe it should just be obj?

Reply:
Good point, that should just be obj. Fixed